src/example-types/metricGroups/dateIntervalGroupStats.js

```diff
@@ -1,22 +1,64 @@
-let moment = require('moment-timezone')
+let moment = require('moment')
 let { groupStats } = require('./groupStatUtils')
 
 let drilldown = ({ field, interval, drilldown }) => {
+  if (interval === 'federalFiscalYear') interval = 'year'
+  if (interval === 'federalFiscalQuarter') interval = 'quarter'
   let gte = drilldown
   let lte = moment
     .parseZone(drilldown)
     .endOf(interval)
     .format()
   return { range: { [field]: { gte, lte } } }
 }
-let buildGroupQuery = ({ field, interval = 'year' }, children, groupsKey) => ({
-  aggs: {
-    [groupsKey]: {
-      date_histogram: { field, interval, min_doc_count: 0 },
-      ...children,
+
+let isOffsetDates = node => {
+  if (
+    node.interval === 'federalFiscalYear' ||
+    node.interval === 'federalFiscalQuarter'
+  )
+    return true
+  return false
+}
+
+let buildGroupQuery = (node, children, groupsKey) => {
+  let { field, interval = 'year' } = node
+
+  // Federal fiscal year quarters have the
+  // same start and end dates as conventional
+  // calendar year quarters but offset forward by one.
+  // e.g. calendarYear2022Q1 => federalFiscalYear2022Q2
+  //      calendarYear2022Q4 => federalFiscalYear2023Q1
+  if (interval === 'federalFiscalYear') interval = 'year'
+  if (interval === 'federalFiscalQuarter') interval = 'quarter'
```

Comment:
we can simplify this to just strip `federalFiscal` off interval in one line (or just `fiscal` if we make the above change)

Reply:
and also factor it out and reuse in `drilldown`